### DISCUSSION TITLE
Which `/_next` routes run `getInitialProps` in `_app.jsx`?

### DISCUSSION THREAD

**Discussion by kepeterson** [vercel/next.js#17426] [Help]
I have a `getInitialProps` in my `_app.jsx` with a log statement in it. Sometimes I see that log statement run when fetching URLs like `/_next/static/j8xtWpNxjQBXgbNyLsf8X/_buildManifest.js` or `/_next/static/chunks/449389e8ba5615083ea6dca7cb8d9efd3d08be8b.998c60c6ad2a173da42b.js` but the vast majority of the time I do not. It is inconsistent. Looking at `_buildManifest.js` specifically, there's not much at all going on in that .js file yet I still see `getInitialProps` in `_app.jsx` being run for some of the requests for it.

For which routes is `getInitialProps` run? Is there a way to tell based on the URL if `getInitialProps` will be run or not for a given request?

I'm on next `9.5.1`.

**Comment by leerob** [comment]
Out of curiosity, what are you using `getInitialProps` for? It's hard to diagnose more without that piece. Do you have a reproducible example by chance? 🙏

**Comment by kepeterson** [comment]
So the actual set up is more sophisticated than just a log statement. I run a custom nextjs server. The custom server handles authentication and sets a `user` property on `req` before passing it into `nextApp.getRequestHandler()`. In `getInitialProps`, I check for `ctx.req` and if it's there, I know I'm doing a server render and I pull out `ctx.req.user` to instantiate my app. This `user` value is then stored in a context so I can access it again on subsequent uses of `next/router` without having to worry about `getInitialProps`.

In the authentication step, running a DB query for the full user is an expensive operation and I'd like to only run it for routes that I know are going to call `getInitialProps` in `_app.jsx`. I haven't been able to determine a completely correct pattern for which routes need to render `_app.jsx` and which do not however. The code below works 99% of the time, but as I alluded to in the OP, sometimes I see errors from routes like `_buildManifest.js` that are excluded by the rule below. But I don't understand why that particular route would need to render `_app.jsx` at all.

```js
const _needsUserData = path => {
  return path.includes('/pages/') || path.startsWith('/next');
};
```

**Comment by Mathspy** [comment]
We have a very similar setup and the fact Next.js calls getInitialProps for `_next` routes doesn't particularly make much sense as those routes are not rendering any pages? All I'd like to know is _why_ Next.js calls getInitialProps for those routes so that I can anticipate that and deal with what Next.js expects from the `appProps` returned from those calls
